clustercontroller-utils/src/main/java/com/yahoo/vespa/clustercontroller/utils/staterestapi/server/JsonWriter.java

```diff
@@ -85,6 +98,26 @@ public void fillInJson(Map<String, SubUnitList> subUnitMap, JSONObject json) thr
         }
     }
 
+    private static void fillInJson(DistributionStates states, JSONObject json) throws Exception {
+        JSONObject statesJson = new JSONObject();
+        statesJson.put("published", distributionStateToJson(states.getPublishedState()));
+        json.put("distribution-states", statesJson);
+    }
+
+    private static JSONObject distributionStateToJson(DistributionState state) throws Exception {
+        JSONObject result = new JSONObject();
+        result.put("baseline", state.getBaselineState());
+        JSONArray bucketSpacesJson = new JSONArray();
+        result.put("bucket-spaces", bucketSpacesJson);
```

Comment:
Feels cleaner to me if this were moved after the loop which sets the elements in `bucketSpacesJson`